app/main/views/send.py

```diff
@@ -606,17 +617,51 @@ def check_messages(service_id, template_id, upload_id, row_index=2):
         metadata_kwargs["sender_id"] = session["sender_id"]
 
     set_metadata_on_csv_upload(service_id, upload_id, **metadata_kwargs)
-
     return render_template("views/check/ok.html", **data)
 
 
+@main.route(
+    "/services/<uuid:service_id>/<uuid:template_id>/check/<uuid:upload_id>/preview",
+    methods=["GET"],
+)
+@main.route(
+    "/services/<uuid:service_id>/<uuid:template_id>/check/<uuid:upload_id>/preview/row-<int:row_index>",
```

Comment:
I'm uncertain if we need this route. It seems like it doesn't matter when I upload a csv with only one row, but I put it there as "just in case". I put it there as a catch if we do need the row index if user uploads csv file that has multiple rows.